presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -168,16 +169,19 @@ public NodeSelector createNodeSelector(ConnectorId connectorId, int maxTasksPerS
         return () -> {
             ImmutableMap.Builder<String, InternalNode> byNodeId = ImmutableMap.builder();
             ImmutableSetMultimap.Builder<HostAddress, InternalNode> byHostAndPort = ImmutableSetMultimap.builder();
+            ImmutableSetMultimap.Builder<HostAddress, InternalNode> inactiveNodesByHostAndPort = ImmutableSetMultimap.builder();
             ImmutableSetMultimap.Builder<InetAddress, InternalNode> byHost = ImmutableSetMultimap.builder();
             ImmutableSetMultimap.Builder<NetworkLocation, InternalNode> workersByNetworkPath = ImmutableSetMultimap.builder();
 
             Set<InternalNode> nodes;
+            Set<InternalNode> inactiveNodes;
             if (connectorId != null) {
                 nodes = nodeManager.getActiveConnectorNodes(connectorId);
             }
             else {
                 nodes = nodeManager.getNodes(ACTIVE);
             }
+            inactiveNodes = nodeManager.getNodes(INACTIVE);
```

Comment:
I'm not sure if it is correct to use INACTIVE here. If we look into `DiscoveryNodeManager`, INACTIVE means version mismatch. What we need is to memoize the dead, crashed, or drained nodes a some period (e.g., 1h).

Reply:
Oh damn, looks like the version here is the presto version and it probably will remain the same :(